### PR TITLE
Apply upstream `aarch64` consecutive register fix (11.x)

### PR DESCRIPTION
**Pull request by jakirkham**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

xref: https://github.com/conda-forge/numba-feedstock/pull/94#issuecomment-1218068344

xref: https://reviews.llvm.org/D104123

cc @gmarkall

### CODE REVIEW

**Comment by jakirkham** [comment]
@conda-forge/llvmdev, this is ready for review

**Comment by h-vetinari** [comment]
There are some failed tests on osx, is that expected...?

File: .azure-pipelines/azure-pipelines-osx.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 jobs:
 - job: osx
   pool:
-    vmImage: macOS-10.15
+    vmImage: macOS-11
```
Comment (jakirkham) [inline .azure-pipelines/azure-pipelines-osx.yml:8]: Moving to macOS 11 may cause issues for LLVM 11's various library tests, which is what seems to be failing

**Review by jakirkham — COMMENTED** [review]
Right think the macOS failures are related to the move to macOS 11. Not sure we can fix those without...

* Reverting to macOS 10.15
* Disabling those tests (not sure how involved this is)
* ...?

**Comment by jakirkham** [comment]
Should add we could also just ignore that job. There are already macOS packages for LLVM. The fix here is focused on ARM, which seems to build ok.

**Comment by hmaarrfk** [comment]
I'm trying to downgrade the compiler in https://github.com/conda-forge/llvmdev-feedstock/pull/167

**Comment by hmaarrfk** [comment]
See https://github.com/conda-forge/llvmdev-feedstock/pull/167#pullrequestreview-1079581882

**Comment by jakirkham** [comment]
Reverting to the old macOS version fixes CI

However @isuruf noted in this comment ( https://github.com/conda-forge/llvmdev-feedstock/pull/165#pullrequestreview-1079581576 ) and following discussion that there is an ABI breakage caused by the patch that would need to be fixed

**Comment by gmarkall** [comment]
@jakirkham In my branch https://github.com/gmarkall/llvmdev-feedstock/tree/fix_llvm_11_aarch I have the consecutive registers patch and the ABI fix applying successfully when I run `build-locally.py` - would you like to take a look at this branch / try it out?

**Comment by hmaarrfk** [comment]
Thanks! I just started the bots and made them work in: https://github.com/conda-forge/llvmdev-feedstock/pull/170

**Comment by gmarkall** [comment]
Looks like it appeared to be successful in #170? @jakirkham Is there anything else you need from me to help move forward right now?

**Comment by jakirkham** [comment]
Resolved by PR ( https://github.com/conda-forge/llvmdev-feedstock/pull/171 ). Closing...

**Pull request closed without merging**